src/fldeff_flash.c

```diff
@@ -28,38 +29,38 @@ struct FlashStruct
 // static functions
 static void hm2_flash(void);
 static void sub_81371B4(void);
-static bool8 sub_8137304(void);
-static void sub_81373F0(void);
-static void sub_8137404(u8 taskId);
-static void sub_8137420(u8 taskId);
-static void sub_81374C4(u8 taskId);
-static void sub_813750C(u8 taskId);
-static void sub_8137574(u8 taskId);
-static void sub_81375A8(void);
-static void sub_81375BC(u8 taskId);
-static void sub_81375D8(u8 taskId);
-static void sub_8137678(u8 taskId);
-static void sub_81376DC(u8 taskId);
+static bool8 MaybeDoMapTransition(void);
+static void DoExitCaveTransition(void);
+static void Task_ExitCaveTransition1(u8 taskId);
+static void Task_ExitCaveTransition2(u8 taskId);
+static void Task_ExitCaveTransition3(u8 taskId);
+static void Task_ExitCaveTransition4(u8 taskId);
+static void Task_ExitCaveTransition5(u8 taskId);
+static void DoEnterCaveTransition(void);
+static void Task_EnterCaveTransition1(u8 taskId);
+static void Task_EnterCaveTransition2(u8 taskId);
+static void Task_EnterCaveTransition3(u8 taskId);
+static void Task_EnterCaveTransition4(u8 taskId);
 
 // rodata
-static const struct FlashStruct gUnknown_085B27C8[] =
+static const struct FlashStruct sTransitionTypes[] =
 {
-    {1, 4, 1, 0, sub_81375A8},
-    {2, 4, 1, 0, sub_81375A8},
-    {3, 4, 1, 0, sub_81375A8},
-    {5, 4, 1, 0, sub_81375A8},
-    {6, 4, 1, 0, sub_81375A8},
-    {7, 4, 1, 0, sub_81375A8},
-    {8, 4, 1, 0, sub_81375A8},
-    {9, 4, 1, 0, sub_81375A8},
-    {4, 1, 0, 1, sub_81373F0},
-    {4, 2, 0, 1, sub_81373F0},
-    {4, 3, 0, 1, sub_81373F0},
-    {4, 5, 0, 1, sub_81373F0},
-    {4, 6, 0, 1, sub_81373F0},
-    {4, 7, 0, 1, sub_81373F0},
-    {4, 8, 0, 1, sub_81373F0},
-    {4, 9, 0, 1, sub_81373F0},
+    {MAP_TYPE_TOWN,        MAP_TYPE_UNDERGROUND, 1, 0, DoEnterCaveTransition},
```

Comment:
Since fields 2 and 3 are annotated as Boolean types, perhaps use the TRUE and FALSE defines here instead of 1 and 0 literals.